### DISCUSSION TITLE
Feature Request: Playhead Tracking Centered on Waveform

### DISCUSSION THREAD

**Discussion by tiimonn** [SubtitleEdit/subtitleedit#9009] [Ideas]
### Feature Request: Playhead Tracking Centered on Waveform

![Waveform example](https://github.com/user-attachments/assets/3d13eb46-7e9d-4b58-b4b8-a76785fc3c92)

**Request:**  
It would be highly beneficial to have an option to keep the playhead (current playtime cursor) centered on the waveform display during playback. Currently, the playhead remains at the far right edge, which limits the view of the upcoming waveform and makes it harder to anticipate changes in the audio.

**Proposed Solution:**  
Introduce a setting to enable continuous playhead tracking that keeps it centered. This would allow users to see more of the upcoming audio content, making editing or reviewing audio much more efficient.

**Productivity Impact:**  
This feature would streamline workflow by providing a clearer, more balanced view of the waveform. For anyone who needs to monitor audio patterns and make precise adjustments, such a feature could save significant time and reduce the strain of constantly adjusting the view.

**Comment by GrampaWildWilly** [comment]
I read this suggestion & thought, "But that's the way it DOES work."  So I went looking for the button that tells SE to do this and . . . I can't find it.  I could swear there was a button on the waveform or a checkbox or something like that to make the waveform behave in the way our original poster here is requesting as a new feature.  Did it accidentally get lost?  My waveform certainly has the current position steady in the center of the waveform.  As playback progresses, it gives the visual impression that the waveform is scrolling right to left behind the stationary vertical bar in the center of the waveform window.

In any case, I decided to look in my Settings.Xml file & I found this:

`    <WaveformCenter>True</WaveformCenter>`

This appears within a very long section of the settings file that contains a lot of this sort of thing:

`<Name of setting>value</Name of setting>`

I'm not entirely sure the `WaveformCenter` setting needs to appear in a particular location within the settings file.  Maybe somebody else could be more informative on this.  My idea is that you should try adding that line to your Settings.Xml (while SE is not running, of course) & see if that gets you what you want.

Sort of as a separate issue, maybe there's been an omission in the GUI that used to be there long ago, which explains why I have that setting in my Settings.Xml file.  The setting is still valid but the GUI whatsit has been accidentally removed from the product at some unknown date.

**Reply by tiimonn** [reply]
Thanks! I will try to get it to work once I'm home.

**Reply by GrampaWildWilly** [reply]
Hmmm......  I just downloaded SE 4.0.8 beta 186, which is the latest.  It looks like this:

![#01](https://github.com/user-attachments/assets/d78f29e6-b798-47ee-a03b-886bc4b6770e)

That image is kind of small.  Just click it & GitHub will open it in a separate window.  You can then Ctrl-+ to magnify it so you can actually see what's there.  What's there is a tooltip hovering up over the control for centering the vertical bar.  The image is so small because I have my controls undocked & the video window is on my second monitor.  You can also see that the cursor within the waveform is centered.

The center control is there.  I have discarded whatever it was that I was running previously.  I'm pretty sure that control was missing but I can't swear to it.  In any case, it's still there in the latest beta.

**Reply by tiimonn** [reply]
![image](https://github.com/user-attachments/assets/c4559899-39a5-4ad1-b037-d4a949b70091)
I've found it :D 
Was searching in the settings.
Thanks!
